backend/src/v5/utils/quota.js

```diff
@@ -15,38 +15,49 @@
  *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 const { createResponseCode, templates } = require('./responseCodes');
+const { getAllUsersInTeamspace, getSubscriptions } = require('../models/teamspaces');
 const DBHandler = require('../handler/db');
 const config = require('./config');
-const { getSubscriptions } = require('../models/teamspaces');
+const { getInvitationsByTeamspace } = require('../models/invitations');
 const { getTotalSize } = require('../models/fileRefs');
 
 const Quota = {};
 
-const calculateQuota = async (teamspace) => {
+Quota.getQuotaInfo = async (teamspace) => {
 	const subs = await getSubscriptions(teamspace);
-	let quotaSize = 0;
+	let freeTier = true;
+	let dataSize = config.subscriptions?.basic?.data ?? 0;
+	let collaborators = config.subscriptions?.basic?.collaborators ?? 0;
 	let hasExpiredQuota = false;
+	let expiryDate = null;
 
 	Object.keys(subs).forEach((key) => {
 		// paypal subs have a different schema - and no oen should have an active paypal sub. Skip.
 		if (key !== 'paypal') {
-			const { expiryDate, data } = subs[key];
-
-			if (expiryDate && expiryDate < Date.now()) {
+			const { expiryDate: subExpiryDate, data, collaborators: subCollaborators } = subs[key];
+			if (subExpiryDate && subExpiryDate < Date.now()) {
 				hasExpiredQuota = true;
 			} else {
-				quotaSize += data;
+				freeTier = false;
+				dataSize += data;
+
+				if (!expiryDate || subExpiryDate < expiryDate) {
+					expiryDate = subExpiryDate;
+				}
+
+				if (collaborators !== 'unlimited') {
+					collaborators = subCollaborators === 'unlimited' ? 'unlimited' : collaborators + subCollaborators;
+				}
 			}
 		}
 	});
 
-	if (hasExpiredQuota && quotaSize === 0) throw templates.licenceExpired;
+	if (hasExpiredQuota && dataSize === config.subscriptions?.basic?.data) throw templates.licenceExpired;
```

Comment:
Of basic plan isn't configured, dataSize would be `0` and config.subscriptions?.basic?.data would be `undefined`

We should also really be checking collaborators as well - as its possible to have collaborators only license

At this point it might be good enough to just do `hasExpiredQuota && !freetier` ? from what it looks like freeTier is only changed to false if we found a valid sub. (I didn't look too deeply, make sure to double check this logic!)